authentik/providers/signals.py

```diff
@@ -0,0 +1,169 @@
+"""Shared Provider signals"""
+
+from urllib.parse import parse_qs, urlencode, urlparse, urlunparse
```

Comment:
this should be 2 separate signal handlers, one in the oauth2 provider and one in the SAML provider

Reply:
I had them in a shared spot because they use the same iframe stage at the same time. I was concerned about a race condition. I'll give it a shot and if worst comes to worst we could run the stage twice, once as oidc and once as saml